src/tests/integration/layouts.rs

```diff
@@ -49,3 +49,45 @@ pub fn accepts_basic_layout() {
     assert_snapshot!(next_to_last_snapshot);
     assert_snapshot!(last_snapshot);
 }
+
+#[test]
+#[ignore]
```

Comment:
@imsnif I ignored these tests as they pass when they run on their own but fail when they run together with all of the others (locally)

Reply:
This is happening due to multiple threads running and causing a race condition. Try increasing the duration from this line to see if this helps. https://github.com/mosaic-org/mosaic/blob/6f168c5ef235bf743668fa8f58d5d55344887b63/src/tests/fakes.rs#L39

Reply:
Thank you :)

Reply:
@henil The problem here is that the `panic` happens inside a single thread and so does not stop the main one - so the program just hangs. I've tried to use `panic::set_hook` and `process::exit` inside it but that exits the entire test process... Is there a clean way of closing the parent thread form its child just for this test or do you think that it should just be left for unit tests?

Reply:
I am not sure about that too. @imsnif any ideas?